packages/@aws-cdk/aws-ecs/lib/fargate/fargate-service.ts

```diff
@@ -73,11 +96,44 @@ export interface IFargateService extends IService {
 export class FargateService extends BaseService implements IFargateService {
 
   /**
-   * Import a task definition from the specified task definition ARN.
+   * Imports from the specified service ARN.
    */
   public static fromFargateServiceArn(scope: cdk.Construct, id: string, fargateServiceArn: string): IFargateService {
     class Import extends cdk.Resource implements IFargateService {
       public readonly serviceArn = fargateServiceArn;
+      public readonly serviceName = cdk.Stack.of(scope).parseArn(fargateServiceArn).resourceName as string;
+    }
+    return new Import(scope, id);
+  }
+
+  /**
+   * Imports from the specified service attrributes.
+   */
+  public static fromFargateServiceAttributes(scope: cdk.Construct, id: string, attrs: FargateServiceAttributes): IBaseService {
+    if ((attrs.serviceArn && attrs.serviceName) || (!attrs.serviceArn && !attrs.serviceName)) {
+      throw new Error('You can only specify either serviceArn or serviceName.');
+    }
+    const stack = cdk.Stack.of(scope);
+    let name: string;
+    let arn: string;
+    if (attrs.serviceName) {
+      name = attrs.serviceName as string;
+      arn = stack.formatArn({
+        partition: stack.partition,
+        service: 'ecs',
+        region: stack.region,
+        account: stack.account,
+        resource: 'service',
+        resourceName: name,
+      });
+    } else {
+      arn = attrs.serviceArn as string;
+      name = stack.parseArn(arn).resourceName as string;
+    }
+    class Import extends cdk.Resource implements IBaseService {
+      public readonly serviceArn = arn;
+      public readonly serviceName = name;
+      public readonly cluster = attrs.cluster;
```

Comment:
I wonder - can you maybe extract this to a module-private function? It's duplicated between this, and `fromEc2ServiceAttributes`.